sematic/plugins/building/docker_builder.py

```diff
@@ -0,0 +1,496 @@
+    according to a proprietary build configuration specified via a configuration file of
+    the form:
+
+    ```
+    version: <version>
+    base_uri: <base image URI>
+    image_script: <custom image URI script>
+    build:
+        requirements: <requirements file>
+        data: <list of data file globs>
+        src: <list of source file globs>
+    push:
+        registry: <image push registry>
+        repository: <image push repository>
+        tag_suffix: <optional image push tag suffix>
+    ```
+
+    It then launches the target pipeline by submitting its execution to Sematic Server,
+    using the build image to execute `@func`s in the cloud.
+    """
+
+    @staticmethod
+    def get_author() -> str:
+        return SEMATIC_PLUGIN_AUTHOR
+
+    @staticmethod
+    def get_version() -> PluginVersion:
+        return _PLUGIN_VERSION
+
+    def build_and_launch(self, target: str):
+        """
+        Builds a container image and launches the specified target launch script, based on
+        proprietary build configuration files.
+        """
+        image_uri = self._build(target=target)
+        self._launch(target=target, image_uri=image_uri)
+
+    def _build(self, target: str) -> ImageURI:
+        """
+        Builds the container image, returning the image URI that can be used to launch
+        executions.
+        """
+        build_config = _get_build_config(script_path=target)
+        logger.info("Loaded build configuration: %s", build_config)
+
+        base_image_uri = _build_image(build_config=build_config, target=target)
+        logger.info("Using container base image URI: %s", repr(base_image_uri))
+
+        # TODO: configure the docker service connection string in the build file
+        docker_client = docker.from_env()  # type: ignore
+
+        # TODO: pull if it is not available locally
+        image = docker_client.images.get(str(base_image_uri))
+        if image.id != base_image_uri.digest:
+            raise BuildError(
+                f"Digest mismatch for image '{base_image_uri}'; "
+                f"expected: {base_image_uri.digest} got: {image.id}"
+            )
+
+        logger.info("Found base image: '%s'", base_image_uri)
+
+        build_image_uri = _push_image(
+            image=image,
+            image_uri=base_image_uri,
+            push_config=build_config.push,
+            docker_client=docker_client,
+        )
+
+        logger.info("Finished building image: '%s'", repr(build_image_uri))
+
+        return build_image_uri
+
+    def _launch(self, target: str, image_uri: ImageURI) -> None:
+        """
+        Launches the specified user code target, using the specified image.
+        """
+        sys.path.append(os.getcwd())
+        # TODO: revert this overwrite after finishing execution
+        #  promote the `environment_variables` testing fixture to a utility
+        os.environ[CONTAINER_IMAGE_ENV_VAR] = repr(image_uri)
```

Comment:
It seems this won't work with the "multiple base images" feature. Do we plan to support that?

Reply:
This won't be supported in the first version that will be exposed to users.